lisk-dex-electron/src/Utils.js

```diff
@@ -40,6 +40,54 @@ const Keys = (dict) => Object.keys(dict);
 
 const Values = (dict) => Object.values(dict);
 
+const EstimationStatus = {
+  MATCH: 1,
+  PARTIAL_MATCH: 2,
+  NO_MATCH: 3,
+};
+
+Object.freeze(EstimationStatus);
+// considering price unit is same for both buyer and seller, for current application it's in terms of lsk per lsh
+const estimatedReturnsForSeller = (amount, price, bids) => {
+  let estimatedReturns = 0;
+  let status = EstimationStatus.NO_MATCH;
+  const maxAmountCanBeReturned = amount * price;
+  for (const bid of bids) {
+    const remainingAmount = maxAmountCanBeReturned - estimatedReturns;
+    if (price <= bid.price) {
+      if (bid.amount >= remainingAmount) {
+        estimatedReturns += remainingAmount;
+        status = EstimationStatus.MATCH;
+        break;
+      } else {
+        estimatedReturns += bid.amount;
+        status = EstimationStatus.PARTIAL_MATCH;
+      }
+    }
+  }
+  return { estimatedReturns, status };
+};
+
+const estimatedReturnsForBuyer = (amount, price, asks) => {
+  let estimatedReturns = 0;
+  let status = EstimationStatus.NO_MATCH;
+  const maxAmountCanBeReturned = amount * (1 / price);
```

Comment:
This is a similar problem as with the `estimatedReturnsForSeller` function. It does not give the user the best deal - In this case the `best deal` means the lowest ask price. You should do the inverse conversion. Now it's doing `bid amount -> ask amount`, but the conversion should be done for each ask individually by converting `ask amount -> bid amount`.